filesystems/gcs/src/main/scala/cromwell/filesystems/gcs/GcsPathBuilder.scala

```diff
@@ -53,10 +53,21 @@ class GcsPathBuilder(authMode: GoogleAuthMode,
                      options: WorkflowOptions) extends PathBuilder {
   authMode.validate(options)
 
-  protected val storageOptions = StorageOptions.builder()
-    .authCredentials(authMode.authCredentials(options))
-    .retryParams(retryParams)
-    .build()
+  val googleProjectProvided = options.getOrElse("google_project", None)
+
+  protected val storageOptions = googleProjectProvided match {
+    case project: String =>
+      StorageOptions.builder()
+        .authCredentials(authMode.authCredentials(options))
+        .retryParams(retryParams)
+        .projectId(project)
+        .build()
+    case None =>
```

Comment:
I forget why this was here and what the right thing to do was but remember that this caused that funky type error. You don't want to have String and None as your two match types.